### ISSUE TITLE
Killed: console storage move

### ISSUE THREAD

**Issue by ingoj** [friendica/friendica#6778]
When running "console storage move", I get a kill every time (2 out of 2 tries ;)): 

```
# sudo -u www-data bin/console storage move
bin/console: line 10:  6100 Killed                  php "${dir}/console.php" "$@"
```

The directory belongs to www-data:www-data and has ug+rwx. 
Directly running php bin/console.php gives just a "Killed".

**Comment by tobiasd** [comment]
Part of #6263

**Comment by MrPetovan** [comment]
Related discussion about OOM error while moving and its potential solutions: https://friendica.mrpetovan.com/display/ec054ce7-115c-7920-2cde-2bd455487105

**Comment by MrPetovan** [comment]
Is anyone working on this? I can do it if anyone else is on it at the moment.

**Comment by annando** [comment]
Not me.

**Comment by fabrixxm** [comment]
Not me.

**Comment by fabrixxm** [comment]
:D

**Comment by MrPetovan** [comment]
I started working on this, the problem is that we use buffered results by default (even with statements and fetches). The solution is to add the possibility in DBA to execute an unbuffered query. It is made somewhat difficult by the difference between how PDO and mysqli set unbuffered queries (one is a flag on the connection while the other is a flag in the query itself) and the prototype of the `DBA::p()` method that doesn't accept additional flags or arguments.

**Comment by nupplaphil** [comment]
Hm .. This seems to be related to #6776 , because with this change, we could create a second `DBA` instance with the "unbuffered" flag at connection, so we don't have to alter the general `DBA` instance..

**Comment by annando** [comment]
I would rather prefer just an additional command or flag or something else. Since I can imagine that both buffered and unbuffered do have their side effects I strongly advocates for not having this behaviour user controllable. I really can imagine multiple possible side effects because of timing issues in the worker or when storing items at a high frequency.

And BTW: Before doing this work, I would suggest doing some tests if this really is the solution.

This "unbuffered" thing will cost less memory, but I can imagine that it would be much better to just work in chunks (like the post update is doing).

But when we find out that this is the solution, then we could simply add some ```DBA::u(...)``` function that calles ```DBA::p(...)``` and before this is done, it would set the database flag. The other calls (like ```DBA::select(...)```) could simply have this behaviour controllable in their options array.

**Comment by MrPetovan** [comment]
I'm with @annando, the unbuffered option should be chosen for isolated queries we know will benefit from it. Not sure a separate DB connection object is a good idea.

> This "unbuffered" thing will cost less memory, but I can imagine that it would be much better to just work in chunks (like the post update is doing).

You're right, I went ahead of myself by adding the unbuffered option to DBA. maybe a while loop with a LIMIT query would do the trick, but the unbuffered option was a cooler project.

**Comment by nupplaphil** [comment]
**If** we really need custom parameters for the database connection, I don't think that it is a good idea to alter the global connection an restore it, like described with `u()` and `p()` or to alter the method signatures (or to mix it). I think this is really a new connection instance with different parameters (and of course a destruction of this connection after the usage). I think this would be more reasonable than altering parameters for a global connection somewhere in the code.

f.e. we alter the parameter for the global `DBA` connection. The move function  implicit calls `DBA` due to `Logger` calls or similar and uses the altered parameters, which **could** lead to unexpected behaviors.

But I'm with you, that these settings shouldn't be available for admins because of the unknown side effects, they should be used just in case the "code" needs it (all power to the code :D ).

**Comment by annando** [comment]
I'm mostly thinking much simpler. We could simply create a new (private) function that does everything that ```p(...)``` currently is doing - except that is accepts a parameter array. Then the new function ```u(...)``` and ```p(...)``` can call this function with different parameters. 

As you can see in https://secure.php.net/manual/en/mysqlinfo.concepts.buffering.php with "mysqli" this setting is done on a "per query" base (so not set during the creation of the connection) while on "pdo" it's vice versa.

And there is another problem with just another database connection: We do have users on shared hosters who are limiting the number of database connections. When a single process could have two open connections this really could be a massive performance drain. And opening and closing the connection on a "per query" base would also be very bad for the performance.

**Comment by MrPetovan** [comment]
Alright, on my 224951-rows `photo` table, the unbuffered query doesn't trigger any OOM error (I don't have any memory limit on CLI either) but it has been running for the past 40 minutes at least without having gone through a single `fetch` yet.

I'm starting to think a worker task that registers a new one when a couple thousands rows have been processed wouold be more suited than a CLI command.

**Comment by MrPetovan** [comment]
Closed via #6902
